### PR TITLE
Refactor/explore creation

### PR DESCRIPTION
**Pull request by supradeep2819**

### CODE REVIEW

File: core/services/explore_service.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+from googleapiclient.discovery import build
+from os.path import dirname, join
+from core.api import create_new_run
+from core.models import Credential, Destination, OAuthApiKeys, Source, StorageCredentials, Sync, Workspace
+logger = logging.getLogger(__name__)
+class ExploreService:
+    @staticmethod
+    def create_spreadsheet(name:str,refresh_token:str):
+        logger.debug("create_spreadsheet")
+        SCOPES = ['https://www.googleapis.com/auth/spreadsheets', 'https://www.googleapis.com/auth/drive']
```
Comment (gane5hvarma) [inline core/services/explore_service.py:15]: lets define constants outside the class, probably global

File: core/services/explore_service.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                fileId=spreadsheet_id,
+                body={
+                    "role": "writer",
+                    "type": "anyone",
+                    "withLink": True
+                },  
+                fields="id"
+            ).execute()
+
+            spreadsheet_url = f"https://docs.google.com/spreadsheets/d/{spreadsheet_id}"
```
Comment (gane5hvarma) [inline core/services/explore_service.py:47]: lets define base_spreadsheet_url 
and do like this
```
spreadsheet_url = f"{base_spreadsheet_url}/{spreadsheet_id}"
```

File: core/services/explore_service.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            print(f"Spreadsheet URL: {spreadsheet_url}")
```
Comment (gane5hvarma) [inline core/services/explore_service.py:48]: remove print

File: core/services/explore_service.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            return spreadsheet_url
+        except Exception as e:
+            logger.error(f"Error creating spreadsheet: {e}")
+            raise Exception("spreadhseet creationm failed")
```
Comment (gane5hvarma) [inline core/services/explore_service.py:52]: creation spelling

Comment (supradeep2819) [reply core/services/explore_service.py:52]: resolved

File: core/services/explore_service.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+            destination["catalog"] = destination_catalog
+            result = Destination.objects.create(**destination)
+            logger.info(result)
+            return result
+        except Exception as e:
+            logger.error(f"Error creating destination: {e}")
+            raise Exception("unable to create destination")
+        
+    @staticmethod
+    def create_sync(source:object,destination:object,workspace_id):
```
Comment (gane5hvarma) [inline core/services/explore_service.py:138]: workspace_id: str

File: core/services/explore_service.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+            sync_config["destination"] = destination
+            sync_config["workspace"] = Workspace.objects.get(id=workspace_id)
+            sync = Sync.objects.create(**sync_config)
+            return sync
+        except Exception as e:
+            logger.error(f"Error creating sync: {e}")
+            raise Exception("unable to create sync")
+
+    @staticmethod
+    def create_run(request,workspace_id,sync_id,payload):
```
Comment (gane5hvarma) [inline core/services/explore_service.py:161]: data types are missing

File: core/explore_api.py [hunk 7]
```diff
@@ -58,102 +48,29 @@ def create_explore(request, workspace_id,payload: ExploreSchemaIn):
...
+        logger.info(destination.id)
+        #create sync
+        sync = ExploreService.create_sync(source,destination,workspace_id)
+        time.sleep(5)
+        #create run
         payload = SyncStartStopSchemaIn(full_refresh=True)
-        response = create_new_run(request,workspace_id,sync.id,payload)
-        print(response)
-        data["name"] = name
+        ExploreService.create_run(request,workspace_id,sync.id,payload)
```
Comment (gane5hvarma) [inline core/explore_api.py:63]: First we should create explore then we should run the sync

**Pull request merged**